pkg/config/node.go

```diff
@@ -313,17 +315,30 @@ func GetLBConfig(kubeconfigPath string, apiPort, lbPort, statPort uint16, apiVip
 		LbPort:   lbPort,
 		StatPort: statPort,
 	}
+
 	// LB frontend address: IPv6 '::' , IPv4 ''
 	if apiVip.To4() == nil {
 		config.FrontendAddr = "::"
 	}
-
-	backends, err := getSortedBackends(kubeconfigPath)
+	// since:
+	// 1: Keepalived script for Api vip checks HAProxy LB status in addition to localhost-kube-apiserver.
+	// 2: Masters nodes details needed for HAProxy config retrieved from kube-apiserver .
+	// to break the circular dependency we should first read masters details from localhost:kube-apiserver instead of
+	// api-vip:kube-apiserver.
+	// Since master nodes details are used (in unicast mode) in api VRRP instance at keeplived config
+	// we should first retrieve the information from localhost:kube-apiserver (to avoid circular dependency),
+	// and only failover to api-vip:kube-apiserver
+	backends, err := getSortedBackends(kubeconfigPath, localhostKubeApiServerUrl)
 	if err != nil {
-		log.WithFields(logrus.Fields{
-			"kubeconfigPath": kubeconfigPath,
-		}).Error("Failed to retrieve API members information")
-		return config, err
+		log.Infof("An error occurred while trying to read master nodes details from localhost:kube-apiserver: %v", err)
+		log.Infof("Trying to read master nodes details from api-vip:kube-apiserver")
+		backends, err = getSortedBackends(kubeconfigPath, "")
```

Comment:
This seems rather magical. Why does passing "" mean api-vip:kube-apiserver? The casual reader would not know how this works. Is there a more explicit value we could pass?

Reply:
I added a Docstring to getSortedBackends function, hope that now it's more explicit.